`vec_box` doesn't take scopes into account

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
mod inner {
    pub struct S;
}

mod inner2 {
    use crate::inner::S;

    pub fn f() -> Vec<Box<S>> {
        vec![]
    }
}
```

I expected to see this happen: Clippy suggests turning `Vec<Box<S>>` into `Vec<S>`

Instead, this happened: Clippy suggests turning it into `Vec<inner::S>`.

```
error[E0433]: failed to resolve: use of undeclared crate or module `inner`
  --> src/lib.rs:10:23
   |
10 |     pub fn f() -> Vec<inner::S> {
   |                       ^^^^^ use of undeclared crate or module `inner`
Original diagnostics will follow.

warning: `Vec<T>` is already on the heap, the boxing is unnecessary.
  --> src/lib.rs:10:19
   |
10 |     pub fn f() -> Vec<Box<S>> {
   |                   ^^^^^^^^^^^ help: try: `Vec<inner::S>`
   |
   = note: `#[warn(clippy::vec_box)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#vec_box
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (ffa2e7a 2020-10-24)
- `rustc -Vv`:
```
rustc 1.47.0 (18bf6b4f0 2020-10-07)
binary: rustc
commit-hash: 18bf6b4f01a6feaf7259ba7cdae58031af1b7b39
commit-date: 2020-10-07
host: x86_64-unknown-linux-gnu
release: 1.47.0
LLVM version: 11.0
```

@rustbot modify labels: +A-suggestion +L-suggestion-causes-error

--- a/clippy_lints/src/types.rs
+++ b/clippy_lints/src/types.rs
@@ -553,7 +553,7 @@ impl Types {
                                     hir_ty.span,
                                     "`Vec<T>` is already on the heap, the boxing is unnecessary.",
                                     "try",
-                                    format!("Vec<{}>", ty_ty),
+                                    format!("Vec<{}>", snippet(cx, boxed_ty.span, "..")),
                                     Applicability::MachineApplicable,
                                 );
                                 return; // don't recurse into the type

--- a/tests/ui/vec_box_sized.fixed
+++ b/tests/ui/vec_box_sized.fixed
@@ -35,4 +35,18 @@ mod should_not_trigger {
     }
 }
 
+mod inner_mod {
+    mod inner {
+        pub struct S;
+    }
+
+    mod inner2 {
+        use super::inner::S;
+
+        pub fn f() -> Vec<S> {
+            vec![]
+        }
+    }
+}
+
 fn main() {}

--- a/tests/ui/vec_box_sized.rs
+++ b/tests/ui/vec_box_sized.rs
@@ -35,4 +35,18 @@ mod should_not_trigger {
     }
 }
 
+mod inner_mod {
+    mod inner {
+        pub struct S;
+    }
+
+    mod inner2 {
+        use super::inner::S;
+
+        pub fn f() -> Vec<Box<S>> {
+            vec![]
+        }
+    }
+}
+
 fn main() {}

--- a/tests/ui/vec_box_sized.stderr
+++ b/tests/ui/vec_box_sized.stderr
@@ -18,5 +18,11 @@ error: `Vec<T>` is already on the heap, the boxing is unnecessary.
 LL |     struct B(Vec<Vec<Box<(u32)>>>);
    |                  ^^^^^^^^^^^^^^^ help: try: `Vec<u32>`
 
-error: aborting due to 3 previous errors
+error: `Vec<T>` is already on the heap, the boxing is unnecessary.
+  --> $DIR/vec_box_sized.rs:46:23
+   |
+LL |         pub fn f() -> Vec<Box<S>> {
+   |                       ^^^^^^^^^^^ help: try: `Vec<S>`
+
+error: aborting due to 4 previous errors
 

